Userland/Libraries/LibC/getsubopt.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ScopeGuard.h>
+#include <AK/StringView.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+// https://pubs.opengroup.org/onlinepubs/9699919799/functions/getsubopt.html
+int getsubopt(char** option_array, char* const* tokens, char** option_value)
+{
+    if (**option_array == '\0')
+        return -1;
+
+    auto option_string = StringView(*option_array);
+
+    auto possible_comma_location = option_string.find(',');
+    char* possible_comma_location_ptr_or_end_array = const_cast<char*>(option_string.characters_without_null_termination()) + option_string.length();
+    if (possible_comma_location.has_value()) {
+        possible_comma_location_ptr_or_end_array = const_cast<char*>(option_string.characters_without_null_termination()) + possible_comma_location.value();
+    }
+
+    auto possible_equals_char_location = option_string.find('=');
+    char* value_start = possible_comma_location_ptr_or_end_array;
+    if (possible_equals_char_location.has_value()) {
+        value_start = const_cast<char*>(option_string.characters_without_null_termination()) + possible_equals_char_location.value();
+    }
+
+    ScopeGuard ensure_end_array_contains_null_char([&]() {
+        if (*possible_comma_location_ptr_or_end_array != '\0')
+            *possible_comma_location_ptr_or_end_array++ = '\0';
+        *option_array = possible_comma_location_ptr_or_end_array;
+    });
+
+    for (int count = 0; tokens[count] != NULL; ++count) {
+        auto token_stringview = StringView(tokens[count]);
+        if (option_string.substring_view(0, token_stringview.length()).compare(token_stringview) != 0)
```

Comment:
`StringView` should have `==` and `!=`, shouldn't it? That would likely be a lot clearer than bringing the equivalent of C string compares into this. :^)